File: 2018/1/1.cpp
// rishav.io

#include <iostream>
#include <vector>
#include <string>
#include <climits>
#include <algorithm>
#include <cmath>
#include <queue>
#include <stack>
#include <map>
#include <set>

using namespace std;

typedef long long LL;

int main() {

	ios_base::sync_with_stdio(false);
	cin.tie(0);

#ifdef __APPLE__
	freopen("input.txt", "r", stdin);
#endif

	int val;
	int t = 0;
	vector<int> vec;
	while ((cin >> val)) {
		t += val;
		vec.push_back(val);
	}
	cout << t << '\n';
	unsigned int idx = 0;
	t = 0;
	set<int> lookup;
	while (true) {
		t += vec[idx];
		if (lookup.find(t) != lookup.end()) {
			cout << t << '\n';
			break;
		}
		lookup.insert(t);
		idx = (idx + 1) % (vec.size());
	}

}

File: 2018/1/1.py
from itertools import cycle

lookup = set([])
vec = []
tmp = 0
with open("input.txt") as f:
    lines = f.readlines()
    for line in lines:
        vec.append(int(line))
        tmp += vec[-1]
print(tmp)
tmp = 0
for x in cycle(vec):
    tmp += x
    if tmp in lookup:
        print(tmp)
        break
    else:
        lookup.add(tmp)
